zipline/_protocol.pyx

```diff
@@ -508,6 +508,9 @@ cdef class BarData:
             # asset isn't alive
             return False
 
+        if asset.auto_close_date and dt >= asset.auto_close_date:
```

Comment:
Throughout `zipline.data.resample` `is_alive_for_session` is used to signify whether there should be data for a given a session. (Which is used to place `nan`s for sessions which evaluate to `False`.

Also, I believe this particular line will fail for futures between market open and midnight. Need to either get the market_open of the `auto_close_date`, or use the `session_label`.